➜  ~ roslaunch seabot driver.launch
... logging to /home/pi/.ros/log/31297c5e-8eff-11ea-94f8-b827ebe168e8/roslaunch-seabot-17974.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

RLException: environment variable 'HOSTNAME' is not set
The traceback for the exception was written to the log file
➜  ~ cat /home/pi/.ros/log/31297c5e-8eff-11ea-94f8-b827ebe168e8/roslaunch-seabot-17974.log
[roslaunch][INFO] 2020-05-05 20:35:28,447: Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
[roslaunch][INFO] 2020-05-05 20:35:28,479: Done checking log file disk usage. Usage is <1GB.
[roslaunch][INFO] 2020-05-05 20:35:28,480: roslaunch starting with args ['/opt/ros/melodic/bin/roslaunch', 'seabot', 'driver.launch']
[roslaunch][INFO] 2020-05-05 20:35:28,482: roslaunch env is {'LC_NUMERIC': 'fr_FR.UTF-8', 'INSIDE_CAJA_PYTHON': '', 'ROS_DISTRO': 'melodic', 'MATE_DESKTOP_SESSION_ID': 'this-is-deprecated', 'XDG_GREETER_DATA_DIR': '/var/lib/lightdm-data/pi', 'ROS_VERSION': '1', 'PKG_CONFIG_PATH': '/home/pi/seabot/devel/lib/pkgconfig:/opt/ros/melodic/lib/pkgconfig', 'XDG_CURRENT_DESKTOP': 'MATE', 'LC_PAPER': 'fr_FR.UTF-8', 'ROSLISP_PACKAGE_DIRECTORIES': '/home/pi/seabot/devel/share/common-lisp', 'LOGNAME': 'pi', 'XDG_SEAT': 'seat0', 'HOME': '/home/pi', 'XDG_VTNR': '7', 'PATH': '/opt/ros/melodic/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin', 'ROS_PACKAGE_PATH': '/home/pi/seabot/src:/opt/ros/melodic/share', 'CMAKE_PREFIX_PATH': '/home/pi/seabot/devel:/opt/ros/melodic', 'ZSH': '/home/pi/.oh-my-zsh', 'DISPLAY': ':0', 'SSH_AGENT_PID': '1604', 'LANG': 'en_US.UTF-8', 'TERM': 'xterm', 'SHELL': '/usr/bin/zsh', 'XDG_SESSION_TYPE': 'x11', 'XAUTHORITY': '/home/pi/.Xauthority', 'LANGUAGE': 'en_US', 'SESSION_MANAGER': 'local/seabot:@/tmp/.ICE-unix/1072,unix/seabot:/tmp/.ICE-unix/1072', 'SHLVL': '1', 'QT_QPA_PLATFORMTHEME': 'gtk2', 'LD_LIBRARY_PATH': '/home/pi/seabot/devel/lib:/opt/ros/melodic/lib', 'QT_ACCESSIBILITY': '1', 'WINDOWID': '35656145', 'GTK_OVERLAY_SCROLLING': '0', 'ROS_LOG_FILENAME': '/home/pi/.ros/log/31297c5e-8eff-11ea-94f8-b827ebe168e8/roslaunch-seabot-17974.log', 'PYTHONPATH': '/home/pi/seabot/devel/lib/python2.7/dist-packages:/opt/ros/melodic/lib/python2.7/dist-packages', 'ROS_MASTER_URI': 'http://localhost:11311', 'GPG_AGENT_INFO': '/run/user/1000/gnupg/S.gpg-agent:0:1', 'QT_AUTO_SCREEN_SCALE_FACTOR': '0', 'LC_MONETARY': 'fr_FR.UTF-8', 'XDG_SESSION_DESKTOP': 'mate', 'XDG_RUNTIME_DIR': '/run/user/1000', 'LC_ADDRESS': 'fr_FR.UTF-8', 'QT_SCALE_FACTOR': '1', 'SSH_AUTH_SOCK': '/run/user/1000/keyring/ssh', 'VTE_VERSION': '5202', 'ROS_ROOT': '/opt/ros/melodic/share/ros', 'GDMSESSION': 'mate', 'XDG_DATA_DIRS': '/usr/share/mate:/usr/local/share:/usr/share:/var/lib/snapd/desktop', 'XDG_SEAT_PATH': '/org/freedesktop/DisplayManager/Seat0', 'XDG_SESSION_PATH': '/org/freedesktop/DisplayManager/Session0', 'XDG_SESSION_ID': 'c1', 'DBUS_SESSION_BUS_ADDRESS': 'unix:path=/run/user/1000/bus', '_': '/opt/ros/melodic/bin/roslaunch', 'LC_IDENTIFICATION': 'fr_FR.UTF-8', 'DESKTOP_SESSION': 'mate', 'LSCOLORS': 'Gxfxcxdxbxegedabagacad', 'XDG_CONFIG_DIRS': '/etc/xdg/xdg-mate:/etc/xdg', 'GTK_MODULES': 'appmenu-gtk-module:gail:atk-bridge:canberra-gtk-module', 'UBUNTU_MENUPROXY': '1', 'OLDPWD': '/home/pi/seabot', 'LS_COLORS': 'rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:', 'GDM_LANG': 'en_US', 'LC_TELEPHONE': 'fr_FR.UTF-8', 'LC_MEASUREMENT': 'fr_FR.UTF-8', 'PWD': '/home/pi', 'ROS_ETC_DIR': '/opt/ros/melodic/etc/ros', 'ROS_PYTHON_VERSION': '2', 'COLORTERM': 'truecolor', 'LC_NAME': 'fr_FR.UTF-8', 'LC_TIME': 'fr_FR.UTF-8', 'LESS': '-R', 'PAGER': 'less', 'USER': 'pi'}
[roslaunch][INFO] 2020-05-05 20:35:28,482: starting in server mode
[roslaunch.parent][INFO] 2020-05-05 20:35:28,484: starting roslaunch parent run
[roslaunch][INFO] 2020-05-05 20:35:28,486: loading roscore config file /opt/ros/melodic/etc/ros/roscore.xml
[roslaunch][INFO] 2020-05-05 20:35:32,856: Added core node of type [rosout/rosout] in namespace [/]
[roslaunch.config][INFO] 2020-05-05 20:35:32,857: loading config file /home/pi/seabot/src/seabot/launch/driver.launch
[roslaunch][ERROR] 2020-05-05 20:35:34,469: RLException: environment variable 'HOSTNAME' is not set
[roslaunch][ERROR] 2020-05-05 20:35:34,471: The traceback for the exception was written to the log file
[roslaunch][ERROR] 2020-05-05 20:35:34,480: Traceback (most recent call last):
  File "/opt/ros/melodic/lib/python2.7/dist-packages/roslaunch/__init__.py", line 330, in main
    p.start()
  File "/opt/ros/melodic/lib/python2.7/dist-packages/roslaunch/parent.py", line 289, in start
    self._start_infrastructure()
  File "/opt/ros/melodic/lib/python2.7/dist-packages/roslaunch/parent.py", line 238, in _start_infrastructure
    self._load_config()
  File "/opt/ros/melodic/lib/python2.7/dist-packages/roslaunch/parent.py", line 144, in _load_config
    roslaunch_strs=self.roslaunch_strs, verbose=self.verbose)
  File "/opt/ros/melodic/lib/python2.7/dist-packages/roslaunch/config.py", line 462, in load_config_default
    raise RLException(e)
RLException: environment variable 'HOSTNAME' is not set

[rospy.core][INFO] 2020-05-05 20:35:34,482: signal_shutdown [atexit]

